diagnostic.js

```diff
@@ -29,26 +29,72 @@ mongoose.Promise = global.Promise;
 mongoose.connect('mongodb://localhost/mongoose-crud');
 
 /// ADD YOUR CODE BELOW
+const db = mongoose.connection;
+const Movement = require('./models/movement.js')
+const done = function() {  db.close(); };  //close connection
 
-const create = (name, description, startYear, endYear) => {};
+const create = (name, description, startYear, endYear) => {
+  Movement.create({
+    name: name,
+    description: description,
+    "starting year" : startYear,
+    "ending year" : endYear,
+  }).then(function(movement){
+    console.log(movement.toJSON());
+  }).catch(console.error).then(done);
+};
 // Success -> Print new Movement as JSON
 // Failure -> Console.error
 
-const index = () => {};
+const index = () => {
+  let searchParams = {};
+   if(arguments[0] && arguments[1]){
+     let field = arguments[0];
+     let criterion = arguments[1];
+     if (criterion[0] === '/'){  //if criterion is a regular expression.
+       let regex = new RegExp(criterion.slice(1, criterion.length - 1));
+       searchParams[field] = regex;
+     } else {
+       searchParams[field] = criterion;
+     }
+   }
```

Comment:
It was not specified that we'd need to do any filtering in `index`, so all of this is unnecessary. However, it will not prevent the desired behavior either.